### PR TITLE
Jetpack connect install & activation instructions

### PR DESCRIPTION
**Pull request by johnHackworth**
This PR starts to implement the changes in the way we tell the user they need to do some manual operations on their wp-admin to keep going with the jetpack-connect flow.

This is the new design:
image

And this is how this PR make it looks:

image

It obviously lacks some design / assets, so I'm passing the ball to @rickybanister  to complete this :) (rick, maybe we can merge this to master as it is right now (maybe hiding those black squares) and work in a new PR, since the examples are going to be a bunch of code and this PR is already a little large)

How to test
========
1. Go to http://calypso.localhost:3000/jetpack/connect 
2. Enter a site where jetpack is not installed
3. You should see the screen above
4. Reload the page and enter a site where jetpack is inactive
4. You should see this:

image

pong @roccotripaldi & @rickybanister

### CODE REVIEW

File: client/signup/jetpack-connect/index.jsx [hunk 1]
```diff
@@ -152,76 +128,108 @@ const JetpackConnectMain = React.createClass( {
...
+
+	render() {
+		const status = this.getStatus();
+		if ( status === 'notJetpack' && ! this.props.jetpackConnectSite.isDismissed ) {
+			return this.renderInstallInstructions();
+		}
+		if ( status === 'notActiveJetpack' && ! this.props.jetpackConnectSite.isDismissed ) {
+			return this.renderActivateInstructions();
+		}
+//		return this.renderActivateInstructions();
```
Comment (johnHackworth) [inline client/signup/jetpack-connect/index.jsx]: To be removed once this is complete... @rickybanister, I let this here so it's easy to test both screens without having to introduce a real site, just uncomment whatever you need to test and that will be the default first screen in jetpack-connect

**Comment by johnHackworth** [comment]
I added some dummy screencaps from the mockups so we can merge this to master for internal testing while rick works on the real examples:

image

**Comment by oskosk** [comment]
Some observations:
* I'm assuming this will be left to Rick, but I'm seeing the cards' content overflowed like this within a narrow window: image
* I was not able to get to the screen for a site which jetpack deactivated. I get to the same screen for sites without jetpack installed (Deactivated jetpack on my sandbox). Maybe I need to be on a particular Jetpack branch too ?

File: client/signup/jetpack-connect/style.scss [hunk 2]
```diff
@@ -90,3 +101,40 @@
 		width: 100%;
 	}
 }
+
+.jetpack-connect__install {
+
+}
```
Comment (oskosk) [inline client/signup/jetpack-connect/style.scss]: Is this empty class placeholder for Rick's magic or unneeded ?

File: client/signup/jetpack-connect/index.jsx [hunk 3]
```diff
@@ -180,7 +184,7 @@ const JetpackConnectMain = React.createClass( {
 					<p>{ this.translate( 'We need to send you to your WordPress install so you can approve the Jetpack installation. Click the button in the bottom-right corner on the next screen to continue.' ) }</p>
```
Comment (oskosk) [inline client/signup/jetpack-connect/index.jsx]: Not related to this particular PR, but... Is "your WordPress install" correct? I'm thinking "WordPress installation" or "WordPress Plugins admin page". Just wondering, don't really know  if "WordPress install" is an usual expression or not

File: client/signup/jetpack-connect/index.jsx [hunk 4]
```diff
@@ -197,10 +201,79 @@ const JetpackConnectMain = React.createClass( {
 					<p>{ this.translate( 'Your jetpack seems to be inactive! We need to send you to your WordPress install so you can activate it before completing the connection.' ) }</p>
```
Comment (oskosk) [inline client/signup/jetpack-connect/index.jsx]: Not related to this particular PR, either. Maybe Jetpack should be spelled with a capital J

File: client/signup/jetpack-connect/index.jsx [hunk 5]
```diff
@@ -212,27 +285,36 @@ const JetpackConnectMain = React.createClass( {
...
 			</Main>
 		);
+	},
+
+	renderInstallInstructions() {
+		return (
+			<Main className="jetpack-connect-wide">
+				<div className="jetpack-connect__install">
+					<ConnectHeader headerText={ this.translate( 'Ready for installation' ) }
+						subHeaderText={ this.translate( 'We\'ll need to send you to your site dashboard for a few manual steps' ) }
```
Comment (oskosk) [inline client/signup/jetpack-connect/index.jsx]: "site dashboard" for "site's dashboard" maybe

**Comment by oskosk** [comment]
> I was not able to get to the screen for a site which jetpack deactivated. I get to the same screen for sites without jetpack installed (Deactivated jetpack on my sandbox). Maybe I need to be on a particular Jetpack branch too ?

Not an issue. I was using jetpack from a `jetpack-react` directory, so the .com API couldn't detect that the plugin was there.

File: client/signup/jetpack-connect/index.jsx [hunk 6]
```diff
@@ -163,76 +139,106 @@ const JetpackConnectMain = React.createClass( {
...
+						step={ 1 }
+						steps={ 3 } />
+					<div className="jetpack-connect__install-steps">
+						<JetpackInstallStep title={ this.translate( '1. Activate Jetpack' ) }
+							text={ this.translate( 'You need to click this tiny blue \'Activate\' link from your plugins list page.' ) }
+							example={ <JetpackExampleActivate /> } />
+						<JetpackInstallStep title={ this.translate( '2. Connect Jetpack' ) }
+							text={ this.translate( 'Once the plugin is activated you\'ll click this blue \'Connect\' button to complete the connection.' ) }
```
Comment (oskosk) [inline client/signup/jetpack-connect/index.jsx]: The button shown in the image is green. The text states "blue"

File: client/signup/jetpack-connect/index.jsx [hunk 7]
```diff
@@ -163,76 +139,106 @@ const JetpackConnectMain = React.createClass( {
...
+						steps={ 3 } />
+					<div className="jetpack-connect__install-steps">
+						<JetpackInstallStep title={ this.translate( '1. Install Jetpack' ) }
+							text={ this.translate( 'You will be redirected to the Jetpack plugin page on your site\'s dashboard to install Jetpack. Click the blue install button.' ) }
+							example={ <JetpackExampleInstall /> } />
+						<JetpackInstallStep title={ this.translate( '2. Activate Jetpack' ) }
+							text={ this.translate( 'Once the plugin is installed, you\'ll need to click this tiny blue \'Activate\' link from your plugins list page.' ) }
+							example={ <JetpackExampleActivate /> } />
+						<JetpackInstallStep title={ this.translate( '3. Connect Jetpack' ) }
+							text={ this.translate( 'Once the plugin is activated you\'ll click this blue \'Connect\' button to complete the connection.' ) }
```
Comment (oskosk) [inline client/signup/jetpack-connect/index.jsx]: The button shown in the image is green too. The text states "blue"

File: client/signup/jetpack-connect/index.jsx (hunk 5 above)
Comment (johnHackworth) [inline client/signup/jetpack-connect/index.jsx]: I'm not so sure about this... let's make a english native take a look at it

**Comment by johnHackworth** [comment]
I've added some mobile styles.

Iphone 4's res:

image


Ipad mini res:
image

Nexus 6:
image

File: client/signup/jetpack-connect/index.jsx (hunk 6 above)
Comment (johnHackworth) [inline client/signup/jetpack-connect/index.jsx]: good poing! :+1:

File: client/signup/jetpack-connect/index.jsx (hunk 3 above)
Comment (johnHackworth) [inline client/signup/jetpack-connect/index.jsx]: We have an issue around for someone from growth or marketing to review all the texts in here ... I guess most of them are going to be changed

File: client/signup/jetpack-connect/index.jsx (hunk 4 above)
Comment (johnHackworth) [inline client/signup/jetpack-connect/index.jsx]: :+1:

Comment (johnHackworth) [inline client/signup/jetpack-connect/index.jsx]: oops, in fact this has been removed already :D

**Comment by oskosk** [comment]
LGTM !

**Comment by johnHackworth** [comment]
Ok, as @oskosk pointed in slack, there are still things to be reviewed (like texts and probably some design polish by @rickybanister ), but I'm going to merge this so we can start the testing

**Pull request merged**